<!-- Use this launch file for SICK LMS200-style sensors with 0.5 deg angular resolution -->
<launch>
    <arg name="suffix" default=""/> <!-- must be set if running multiple instances -->
    
    <arg name="laser" default="laser"/> <!-- laser topic -->
    <arg name="subscriber_queue_size" default="1"/>
    <arg name="synchronizer_queue_size" default="5"/>

    <arg name="detected_persons" default="detected_persons"/> <!-- output detected persons topic -->
    <arg name="detection_id_increment" default="1"/>
    <arg name="detection_id_offset" default="0"/>
    
    <arg name="segmentation" default="true"/>  <!-- set to false if segmentation is provided externally -->
    <arg name="segmentation_type" default="extended_jump_distance_segmentation"/>
    <arg name="distance_threshold" default="0.1"/>

    <arg name="min_avg_distance_from_sensor"/>
    <arg name="max_avg_distance_from_sensor"/>
    <arg name="min_points_per_segment"/>
    <arg name="max_points_per_segment"/>

    <arg name="min_width" default="0.05"/>
    <arg name="max_width" default="0.5"/>

    <arg name="pose_variance" default="0.01"/>

    <remap from="laser" to="$(arg laser)"/>
    <remap from="detected_persons" to="$(arg detected_persons)"/>

    <node name="laser_segmentation$(arg suffix)" type="$(arg segmentation_type)" pkg="srl_laser_segmentation" if="$(arg segmentation)">
        <param name="min_avg_distance_from_sensor" value="$(arg min_avg_distance_from_sensor)"/>
        <param name="max_avg_distance_from_sensor" value="$(arg max_avg_distance_from_sensor)"/>
        <param name="min_points_per_segment" value="$(arg min_points_per_segment)"/>
        <param name="max_points_per_segment" value="$(arg max_points_per_segment)"/>
        <param name="distance_threshold" value="$(arg distance_threshold)"/>
        <param name="queue_size" value="$(arg subscriber_queue_size)"/>
    </node>

    <node name="laser_detector$(arg suffix)" type="blob_detector" pkg="srl_laser_detectors" args="" output="screen">
        <param name="detection_id_increment" value="$(arg detection_id_increment)"/>
        <param name="detection_id_offset" value="$(arg detection_id_offset)"/>
        <param name="min_width" value="$(arg min_width)"/>
        <param name="max_width" value="$(arg max_width)"/>
        <param name="min_points" value="$(arg min_points_per_segment)"/> <!-- deprecated -->
        <param name="max_points" value="$(arg max_points_per_segment)"/> <!-- deprecated -->
        <param name="max_range" value="$(arg max_avg_distance_from_sensor)"/> <!-- deprecated -->
        <param name="pose_variance" value="$(arg pose_variance)"/>
        <param name="subscriber_queue_size" value="$(arg subscriber_queue_size)"/>
        <param name="synchronizer_queue_size" value="$(arg synchronizer_queue_size)"/>
    </node>

</launch>